core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -358,10 +358,20 @@ public long spill(long numBytes) throws IOException {
           return 0L;
         }
 
+        long released = 0L;
+        // longArray will not be used anymore, but may take significant amount of memory
+        // but never released until the task is complete.
```

Comment:
This line seems to explain the behavior of current code (before applying this PR). It would be good to explain the behavior of the new code.